force-app/main/default/classes/SetPrimaryContactToOpp.cls

```diff
@@ -0,0 +1,20 @@
+public with sharing class SetPrimaryContactToOpp {
+    public static void setPrimaryContactOnAccToOpp(List<Opportunity> oppRecords){
+        Set<Id> accountIds = new Set<Id>();
+        for(Opportunity opp : oppRecords){
+            accountIds.add(opp.AccountId);
+        }
+        List<Opportunity> oppsToBeUpdated = new List<Opportunity>();
+        List<Contact> primaryContacts = [SELECT Id, FirstName, LastName, Title, AccountId, CreatedDate 
+                                        FROM Contact 
+                                        WHERE Title = 'CEO' AND AccountId IN :accountIds];
+        for(Opportunity currentOpp : oppRecords){
+            for(Contact con : primaryContacts){
+                if(con.AccountId == currentOpp.AccountId && con.Title == 'CEO'){
+                    currentOpp.Primary_Contact__c = con.Id;
+                    oppsToBeUpdated.add(currentOpp);
+                }
```

Comment:
Good work solving this. I suggest that this code couold be more performant with a Map. In your solution with the nested for loop, you will evaluate every contact for each iteration of the outer loop. If that list had 1000s of Contacts, checking them every time would be incredibly slow. 

If you had 10 opportunities and 100 contacts. You would iterate over that contact list 10 times for 1000 evaluations. This will probably not scale well. 

```
 private void setPrimaryContact(){
        Set<Id> accountIds = getIdValues(opportunities, Opportunity.AccountId);
        List<Contact> contacts = [SELECT Id, AccountId, Title
        FROM Contact
        WHERE AccountId IN: accountIds AND Title = 'CEO'
        WITH SYSTEM_MODE ];

        Map<Id, Contact> contactsByAccountId = new Map<Id, Contact>();
        for(Contact contact : contacts){
            contactsByAccountId.put(contact.AccountId, contact);
        }
        for(Opportunity opportunity : opportunities){
           if(contactsByAccountId.containsKey(opportunity.AccountId)){
               opportunity.Primary_Contact__c = contactsByAccountId.get(opportunity.AccountId).Id;
           }
        }
    }
    ```